### DISCUSSION TITLE
Substitute the magrittr Pipe `%>%` with R’s Native Pipe Operator `|>` | /en/2022/04/magrittr-native-pipe/

### DISCUSSION THREAD

**Discussion by giscus** [rbind/yihui#1618] [Comments]
R 4.1.0 has added a native pipe operator |&gt; to base R. If you are using the magrittr pipe %&gt;% and want to switch to |&gt; in your R scripts (you do not have to), I have a quick function &hellip;

https://yihui.org/en/2022/04/magrittr-native-pipe/

**Comment by fingitt** [comment]
Nice tool, Yihui! I noticed a few more cases in which it doesn't work:
1. In some places R expects a comma after an item. If you put a line of comment there, it gets converted to invisible(...) without a trailing comma and breaks the formatter. E.g.:
```
x <- c(
  # a is not used
  a = 0
)
```

2. Sometimes R doesn't accept the ( after invisible, e.g., if you have a comment inside a function declaration like this:
```
x <- function(
  # First argument does nothing
  a = 1
) {
  0
}
```

**Reply by yihui** [reply]
That's the same issue as https://github.com/yihui/formatR/issues/103
